tracer/src/Datadog.Trace/Agent/StatsAggregationKey.cs

```diff
@@ -0,0 +1,60 @@
+// <copyright file="StatsAggregationKey.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+
+namespace Datadog.Trace.Agent
+{
+    internal readonly struct StatsAggregationKey : IEquatable<StatsAggregationKey>
```

Comment:
Could we use a `record` here? The compiler will implement `IEquatable<StatsAggregationKey>`, `GetHashCode()`, and equality operators.

```csharp
internal readonly record struct StatsAggregationKey(
            string Resource,
            string Service,
            string OperationName,
            string Type,
            int HttpStatusCode);
```

Reply:
This seems like a good place for `record`, but Kevin noted that there is at least one minor workaround needed for .NET Framework: https://stackoverflow.com/a/65663416/869621 . For this PR, my preference is to keep this as-is and later we can do a refactoring of `readonly struct`s to `readonly record struct`s